isofit/utils/apply_oe.py

```diff
@@ -321,21 +321,29 @@ def apply_oe(args):
 
     gmtime = float(h_m_s[0] + h_m_s[1] / 60.0)
 
-    # get radiance file, wavelengths
-    if args.wavelength_path:
+    # get radiance file, wavelengths, fwhm
+    radiance_dataset = envi.open(envi_header(paths.radiance_working_path))
+    wl_ds = np.array([float(w) for w in radiance_dataset.metadata["wavelength"]])
+    try:
         chn, wl, fwhm = np.loadtxt(args.wavelength_path).T
-    else:
-        radiance_dataset = envi.open(envi_header(paths.radiance_working_path))
-        wl = np.array([float(w) for w in radiance_dataset.metadata["wavelength"]])
+        if len(chn) != len(wl_ds) or not all(wl == wl_ds):
+            logging.warning(
+                "Number of channels or center wavelengths provided in wavelength file do not match"
+                " wavelengths in radiance cube. Adopting center wavelengths from ENVI"
+                " header."
+            )
+            raise ValueError
+    except ValueError or FileNotFoundError:
+        wl = wl_ds
```

Comment:
My personal view is that this is an unsafe operation.  If we have multiple different wavelengths being passed that don't align, I think we need to throw a hard error and make the user fix it - this type of soft warning is too easy to overlook.  Open to other views here, but that's my initial thought.

Reply:
I agree! I was basically open to any thoughts here. Wasn't sure if we should always fall back to the wavelength grid from the radiance cube by default. Do you think a hard error is necessary for both cases, i.e., `len(chn) != len(wl_ds)` and `not all(wl == wl_ds)`? The former in any case I'd say.

Reply:
I think it's gotta be both, though we could change to a np.isclose instead of an equals, with a tolerance of maybe 0.01 or 0.001 nm?

Reply:
See [9fa6f3e](https://github.com/isofit/isofit/pull/476/commits/9fa6f3e3e4dc514819c391ce704b88c60059d0a6).